src/virtualenv/discovery/cached_py_info.py

```diff
@@ -103,6 +103,10 @@ def _run_subprocess(cls, exe):
     with ensure_file_on_disk(resolved_path) as resolved_path:
         cmd = [exe, "-s", str(resolved_path)]
 
+        # Prevent sys.prefix from leaking into the child process on macOS.
+        if sys.platform == "darwin":
```

Comment:
I have a feeling that we should do this unconditionally 🤔 that being said I'm not a fan of modifying the os.environ directly; instead copy it, modify it and then pass it into the Popen

Reply:
we should also add some test; and changelog for this

Reply:
oh and do link the upstream bug ticket as a comment 👍